pkg/apis/ark/v1/volume_snapshot.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshot is... TODO
+type VolumeSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   VolumeSnapshotSpec   `json:"spec"`
+	Status VolumeSnapshotStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshotList is a list of VolumeSnapshots.
+type VolumeSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []VolumeSnapshot `json:"items"`
+}
+
+// VolumeSnapshotSpec defines the specification for an Ark VolumeSnapshot.
+type VolumeSnapshotSpec struct {
+	// Provider is the provider of the backup storage.
+	Provider string `json:"provider"`
+
+	// Config is for provider-specific configuration fields.
+	Config map[string]string `json:"config"`
+
+	// SnapshotID is the ID of the snapshot taken in the cloud
+	// provider API of this volume.
+	SnapshotID string `json:"snapshotID"`
+
+	// Type is the type of the disk/volume in the cloud provider
+	// API.
+	Type string `json:"type"`
+
+	// AvailabilityZone is the where the volume is provisioned
+	// in the cloud provider.
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// Iops is the optional value of provisioned IOPS for the
+	// disk/volume in the cloud provider API.
+	Iops *int64 `json:"iops,omitempty"`
+}
+
+// VolumeSnapshotStatus captures the current status of an Ark VolumeSnapshot.
+type VolumeSnapshotStatus struct {
+	// Version is the backup format version.
+	Version int `json:"version"`
+
+	// Expiration is when this Backup is eligible for garbage-collection.
+	Expiration metav1.Time `json:"expiration"`
```

Comment:
this shouldn't be needed as it'll be driven from the overall Backup's expiration